### DISCUSSION TITLE
product name guessing

### DISCUSSION THREAD

**Discussion by TuTAH1** [cvzi/Metacritic-userscript#10] [Ideas]
Hello! I think it would be nice to implement this feature:
If script doesn't find the product by name it takes from product.data, it try to "guess" it that way:
1. Deleting the text in the brackets ([Example](https://store.steampowered.com/app/295790/Never_Alone_Kisima_Ingitchuna/))
2. Deleting all non-letter symbols ( like ©, ®, etc.)
3. Deleting "-" and all text after it. ([Example](https://store.steampowered.com/app/751630/After_the_Fall__Launch_Edition/))

And try to search for a product with a new name. What do you think?

**Comment by cvzi** [comment]
If think I could add a function that is called if there were no results, so the function can broaden the search.
Something like this:
```javascript
{
  steam: {
    host: ['store.steampowered.com'],
    condition: () => document.querySelector('*[itemprop=name]'),
    products: [{
      condition: Always,
      type: 'pcgame',
      data: () => document.querySelector('*[itemprop=name]').textContent,
      broaden: function(lastProductData) {
        // Remove brackets from search term
        return removeBrackets(lastProductData)
      }
    }]
  }
}
```

**Reply by TuTAH1** [reply]
I think it would be better to put it at a higher level, than host-specific. Previously I think to add it globally (so, if it can't find result in any site, it will execute this function), but now I think... to add it globally too, but type-specific filter (`switch (type)`). Or maybe even some of filters would be global, some – type-specific. I'm just not sure where's the code that makes a window after the result is found.

**Comment by cvzi** [comment]
I have added it on a global level:
https://github.com/cvzi/Metacritic-userscript/blob/9ec60d53830a319ae10ac391fa218e9d1530d56b/Show_metacritic_ratings.user.js#L320-L346

For now the global function only applies to 'pcgame' and it can do the things that you suggested. It goes through three steps, and tries to search after each step. I thought that might be better than removing everything at once.

I also added the product specific way as described before. So it is possible to overwrite the global function in the product with `broaden: function(data, step, type) {...\` 

I have not really tested it yet, that's why I have not increased the version number. If you want to test it, you have to manually install it from the github file.
